lib/lists.nix

```diff
@@ -94,7 +94,14 @@ rec {
 
      Type: foldl' :: (b -> a -> b) -> b -> [a] -> b
   */
-  foldl' = builtins.foldl' or foldl;
+  foldl' =
+    f: acc: list:
+    # The builtin `foldl'` is a bit lazier than expected, see https://github.com/NixOS/nix/pull/7158.
+    # In particular, the initial accumulator value is not forced before the iteration starts.
```

Comment:
```suggestion
    # In particular, the initial accumulator value is not forced before the first iteration starts.
```

Reply:
I think we should put this in the user facing documentation:

```
Unlike `builtins.foldl'`, the initial accumulator is forced before the first iteration.
```

Unfortunately we haven't defined the accumulator.

Reply:
I rewrote the `lib.foldl'` documentation now to include this and more.